dev/tests/agent.rs

```diff
@@ -268,7 +268,11 @@ struct TestFixture {
     agent: Agent,
 }
 
-async fn do_spawn_request(mut request: SpawnRequest) -> (TestFixture, BackendStateSubscription) {
+async fn do_spawn_request(
```

Comment:
I know this pre-existed this PR, but maybe we should rename this `spawn_test_backend` to be more accurate now that it has arguments that relate to the specific backend it spawns.

Reply:
was a mistake to put this stuff here, I removed it.